merger/fix.go

```diff
@@ -57,16 +58,30 @@ func FixLoads(f *rule.File, knownLoads []rule.LoadInfo) {
 	}
 
 	// Make a map of all the symbols from known files used in this file.
-	usedKinds := make(map[string]map[string]bool)
-	for _, r := range f.Rules {
-		kind := r.Kind()
-		if file, ok := knownKinds[kind]; ok && !otherLoadedKinds[kind] {
-			if usedKinds[file] == nil {
-				usedKinds[file] = make(map[string]bool)
+	usedSymbols := make(map[string]map[string]bool)
+	bzl.Walk(f.File, func(x bzl.Expr, stk []bzl.Expr) {
```

Comment:
I am pretty sure the following logic would lead to less iteration, unless I am missing the reason why you need to loop through `stk`:
```
bzl.Walk(f.File, func(x bzl.Expr, stk []bzl.Expr) {
  ce, ok := x.(*bzl.CallExpr)
  if !ok {
    return
  }
  ...
}

Reply:
Good call - updated!